compiler/luci-interpreter/src/kernels/SpaceToDepth.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SpaceToDepth.h"
+#include "Utils.h"
+#include <tensorflow/lite/kernels/internal/optimized/optimized_ops.h>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+SpaceToDepth::SpaceToDepth(const Tensor *input, Tensor *output, const SpaceToDepthParams &params)
+    : KernelWithParams<SpaceToDepthParams>(params), _input(input), _output(output)
+{
+}
+
+void SpaceToDepth::configure()
+{
+  assert(_input->shape().num_dims() == 4);
+  assert(_output->element_type() == DataType::FLOAT32 || _output->element_type() == DataType::U8 ||
+         _output->element_type() == DataType::S8 || _output->element_type() == DataType::S32 ||
+         _output->element_type() == DataType::S64);
+  assert(_input->element_type() == _output->element_type());
+
+  const int block_size = params().block_size;
+  const int input_height = _input->shape().dim(1);
+  const int input_width = _input->shape().dim(2);
+  int output_height = input_height / block_size;
+  int output_width = input_width / block_size;
```

Comment:
(nitpick) I would use `int32_t` for values related to dimension size, just for consistency.

Reply:
okay, I will fix it.